jobs/fxci-taskcluster-export/fxci_etl/metric/export.py

```diff
@@ -87,21 +88,32 @@ def get_timeseries(self, project: str, interval: TimeInterval):
         return results
 
     def get_time_interval(self) -> TimeInterval:
+        """Return the time interval to query metrics over.
+
+        This will grab metrics all metrics from the last end time, up until
+        11:59:59 of yesterday. Ideally the metric export runs in a daily cron
+        task, such that it exports a days worth of data at a time.
+        """
+        utc = pytz.UTC
+        now = datetime.now(utc)
+        yesterday = now.date() - timedelta(days=1)
+        end_time = utc.localize(datetime.combine(yesterday, datetime.max.time()))
```

Comment:
The approach you're taking here is to handle execution date logic yourself. This will work, but we usually rely on a date parameter passed in from Airflow and make the job run idempotent (which IIUC is also the case here because you store the state in GCS).
To do this in a more standard way, I'd pass a `date` parameter from Airflow like in [this job](https://github.com/mozilla/telemetry-airflow/blob/main/dags/dap_collector.py#L64) and use it to process all data from that day. This way I'd be sure that `date` is a "complete" day and this would allow to simplify this function.